components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatement.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.store.mongodb.query;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+import com.mongodb.DBRef;
+import com.mongodb.WriteResult;
+import com.mongodb.DBCursor;
+import com.mongodb.AggregationOutput;
+import com.mongodb.WriteConcern;
+import com.mongodb.DBEncoder;
+import com.mongodb.BulkWriteResult;
+import org.bson.types.BSONTimestamp;
+import org.bson.types.Binary;
+import org.bson.types.Symbol;
+import org.wso2.carbon.user.api.UserStoreException;
+
+/**
+ * MongoDB Prepared Statement Interface
+ */
+@SuppressWarnings({"unused", "UnusedReturnValue"})
+public interface MongoPreparedStatement {
+
+    /**
+     * set int parameter value to respective query parameter

```

Comment:
Capitalize first letter class level and method level comments and end with a period.